sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c

```diff
@@ -179,8 +179,10 @@ int gnrc_pktbuf_realloc_data(gnrc_pktsnip_t *pkt, size_t size)
         pkt->data = new_data;
     }
     else {
-        _pktbuf_free(((uint8_t *)pkt->data) + aligned_size,
-                     pkt->size - aligned_size);
+        if (_align(pkt->size) > aligned_size) {
```

Comment:
Forget what I said in https://github.com/RIOT-OS/RIOT/pull/4544#discussion_r49029179. Can you maybe provide a test case for this error case?